src/nvim/buffer.c

```diff
@@ -5493,28 +5497,98 @@ static int buf_signcols_inner(buf_T *buf, int maximum)
   return signcols;
 }
 
+/// Calculate whether buf->b_signcols needs to be invalidated after deleting
```

Comment:
I'd formulate this differently. "Calculate" to me sounds like it (just) returns a value, but we are actually changing state here. Perhaps something more like "Invalidate buf->b_signcols if needed, after deleting signs between line1 and line2 (inclusive)"